.autorc

```diff
@@ -3,5 +3,13 @@
     "baseBranch": "master",
     "author": "auto <auto@nil>",
     "noVersionPrefix": true,
-    "plugins": ["git-tag"]
+    "plugins": [
+        "git-tag",
+        [
+            "exec",
+            {
+                "afterChangelog": "bump2version \"$(printf '%s\n' \"$ARG_0\" | jq -r .bump)\""
```

Comment:
what is the benefit of bringing in bump2version over some simple `echo "__version__ = \"$ARG_0\"" >|  datalad_crawler/version.py` if `ARG_0` is the one containing version here?

Reply:
`$ARG_0` doesn't contain the new version; it contains the name of the version level that is being incremented ("major", "minor", or "patch"), which is perfect for `bump2version`.

Reply:
and auto itself has no functionality to bump or report back the version? I thought it is the one which "mints" a new version to be, since otherwise I do not see e.g. how in dandi-cli version is decided upon -- we rely on versioneer, and I thought that one just uses the git tag, but then something mints the git tag (so the version).  could you please digest the workflow here for me a bit more?

Reply:
auto can and does determine a new version, but (last time I checked) its `$ARG_0` payload does not include that information; it only contains the changed version level.  auto determines the version level based on the tags on the PRs since the last release, increments the version of the last release appropriately, generates & commits a changelog, runs the `afterChangelog` hook here to increment `__version__` in the source code and commit the change, and then it tags the latest commit with the new version, pushes the tag, and creates a GitHub release.

Even if auto did report the actual version, the `afterChangelog` command would still need to include a `git commit` command after updating `version.py`, whereas `bump2version` can do all of that at once.

Reply:
THANK YOU.  Now it feels less magical.  I would still prefer to avoid bump2version: even if both `auto` and `bump2version` use the similar logic for bumping a version, that is two different logics and might result in different results.  Also still feels too heavy for such a tiny job.
If I read https://github.com/intuit/auto/blob/main/packages/core/src/auto.ts#L1880 code correctly, `ARG_4` might be the `currentVersion` which if I get it right - is the new version `auto` decided upon.  I think it is Ok to do a commit manually BUT ideally it should even be done along with the commit to the changelog: and looking at https://github.com/intuit/auto/blob/main/packages/core/src/auto.ts#L1896 it seems that `beforeCommitChangelog` should be the hook to use to produce desired version.py.  So could we 'tap' into that hook?

Even more ideally -- name of the package should be figured out automagically and it is possible!:

```shell
$> python setup.py --name 2>/dev/null
datalad_crawler
```
This would allow for the release workflow to be easily reused across multiple packages.

Reply:
I configured auto in a test repository to dump the environment in the `beforeCommitChangelog` and `afterChangelog` hooks; you can see the results [here](https://github.com/jwodder/auto-test/tree/master/env).  As you can see, the only `$ARG_x` variable is `$ARG_0`, and the `currentVersion` field within the JSON value is the version *before* incrementing it for the new release.  Hence, we need to either keep using bump2version or else duplicate its logic.

Reply:
ah -- ARG_0 is a json record, so you just need (copied into clipboard, in your case it is the `$ARG_0`)
```
$> xsel -o | jq '.currentVersion'  
"0.3.1"
```
?

Reply:
Except that `currentVersion` is the *previous* version, not the new version that is being released.

Reply:
note that both
```
  "lastRelease": "0.3.1",
  "currentVersion": "0.3.1"
```
are identical for that run.  was 0.3.1 the release to come?

Reply:
No, as I said, the env dumps were generated when releasing version 0.3.2.

Reply:
See also: [this comment](https://github.com/intuit/auto/issues/1266#issuecomment-636215261)

Reply:
ah, that comment did not appear whenever I made my comment. gotcha.  
it is a bit odd then that `auto` does not provide updated version - what is the point of having currentVersion and lastRelease to match?  
 Is there may be some `auto` command which would print an updated version?

Reply:
worse comes to worse, it could as well be
```
awk 'NR==1{print $2}' CHANGELOG.md
```
to pick up the version, in `beforeCommitChangelog.env` right?

Reply:
may be with some care/test with grep that obtained value is a semver in syntax